### PR TITLE
Vet Clinic - Multiple tables, Relational database usage

### PR DESCRIPTION
**Pull request by jmontanholi**
# In this feature branch I implemented the following:

* ## Database schema.
   * ### Animals table.
     * #### Inserted species_id inside the Animals table as foreign key.
     * #### Inserted owner_id inside the Animals table as foreign key.
   * ### Species Table.
     * #### Inserted id as primary key.
     * #### Inserted name property.
   * ### Owners Table. 
     * #### Inserted id as primary key.
     * #### Inserted full_name property.
     * #### Inserted age property. 
      
  * ### Created queries that select specific information from the relational database schema.
  * ### Used transitions to safely update the tables.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -2,11 +2,32 @@
...
+  owner_id INT,
+  CONSTRAINT species_id
+    FOREIGN KEY(species_id)
+      REFERENCES species(id)
+      ON DELETE CASCADE,
+  CONSTRAINT owner_id
+    FOREIGN KEY(owner_id)
+      REFERENCES owners(id)
+      ON DELETE CASCADE
 );
```
Comment (Emmyn5600) [inline schema.sql:33]: - [ ] You are missing the following:

 You should modify the animals table like this:
 
 - Remove column species
 - Add column species_id which is a foreign key referencing species table
 - Add column owner_id which is a foreign key referencing the owners table

File: data.sql [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+INSERT INTO owners
+              (full_name, age)
+            VALUES
+              ('Sam Smith', 34), 
+              ('Jennifer Orwell', 19), 
+              ('Bob', 45), 
+              ('Melody Pond', 77), 
+              ('Dean Winchester', 14),
+              ('Jodie Whittaker', 38);
+
```
Comment (Emmyn5600) [inline data.sql]: - [x] I see this way should work however, You are required to implement like this:

 INSERT INTO owners (full_name, age)
 VALUES ('Sam Smith', 34);

 INSERT INTO owners (full_name, age)
 VALUES ('Jennifer Orwell', 19);

 INSERT INTO owners (full_name, age)
 VALUES ('Bob', 45);

 INSERT INTO owners (full_name, age)
 VALUES ('Melody Pond', 77);

 INSERT INTO owners (full_name, age)
 VALUES ('Dean Winchester', 14);

 INSERT INTO owners (full_name, age)
 VALUES ('Jodie Whittaker', 38);

File: data.sql [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+INSERT INTO species (name) VALUES ('Pokemon'), ('Digimon');
+
+BEGIN TRANSACTION;
+UPDATE animals SET species_id = 2 WHERE name LIKE '%mon';
+UPDATE animals SET species_id = 1 WHERE species_id IS NULL;
```
Comment (Emmyn5600) [inline data.sql]: - [x] If an animal's name ends with mon it should have Digimon as a species_id, all other animals should have Pokemon instead, kindly let us update our animals using names instead.

File: data.sql [hunk 4]
```diff
@@ -1,6 +1,43 @@
...
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
+UPDATE animals SET owner_id = 2 WHERE name = 'Gabumon' OR name = 'Pikachu';
+UPDATE animals SET owner_id = 3 WHERE name = 'Devimon' OR name = 'Plantmon';
+UPDATE animals SET owner_id = 4 WHERE name = 'Charmander' OR name = 'Squirtle' OR name = 'Blossom';
+UPDATE animals SET owner_id = 5 WHERE name = 'Angemon' OR name = 'Boarmon';
+SELECT full_name AS owner, name FROM animals JOIN owners ON animals.owner_id = owners.id;
```
Comment (Emmyn5600) [inline data.sql]: - [x] In this implementation `all animals got assigned owners`, so you need to update while you are selecting id from owners.

Sample answer for Q1
`UPDATE animals
   SET owner_id = (SELECT id from owners WHERE full_name = 'Sam Smith')
   WHERE name = 'Agumon';`

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @jmontanholi 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 

  - Please check the comment under the review.
  
  Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: schema.sql (hunk 1 above)
Comment (jmontanholi) [reply schema.sql:33]: Sorry, this is exactly what it is in the file?

Comment (jmontanholi) [reply schema.sql:33]: https://www.postgresqltutorial.com/postgresql-foreign-key/

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi @jmontanholi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**